app/models/configuration_script.rb

```diff
@@ -1,3 +1,20 @@
 class ConfigurationScript < ActiveRecord::Base
-  belongs_to :configuration_manager, :class_name => 'ManageIQ::Providers::ConfigurationManager'
+  belongs_to  :manager,
+              :class_name  => 'ManageIQ::Providers::ConfigurationManager',
+              :foreign_key => :configuration_manager_id
+  include ProviderObjectMixin
+
+  def run
+    with_provider_object do |jt|
+      jt.launch('extra_vars' => variables)
```

Comment:
Yes, please. extra_vars pulled from the template are default values. Any user modified values need to be passed in. You need to expose this method in its corresponding miq_ae_service model.